WordPress/gutenberg
# test: Enable jest-watch-typeahead for native tests

## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/5904

## What?

Enable `jest-watch-typeahead`, allowing filtering tests by file name or test name
in "watch" mode.

## Why?

File name and test name filtering makes it easier to quickly select tests to
run.

## How?

> **Warning**: This is a breaking change in the form of the package script name changes, e.g. `npm run native test` is changed to `npm run test:native`. 
- test: Enable `jest-watch-typeahead` for native tests
- test: Refactor Jest configuration to fix filter regex
    Previously, Jest was executed from within a nested directory. Doing so
    caused the regex patterns used for the `jest-watch-typeahead` to break,
    as the `../../` would be erroneously included in the file name.

## Testing Instructions




1. Enable Jest watch mode: `npm run test:native -- --watch`
2. Press <kbd>p</kbd> to enable test file name filtering. 
3. Type to begin filtering, select a test file. 
4. Verify only the selected file runs. 
5. Press <kbd>t</kbd> to enable test name filtering. 
6. Type to begin filtering, select a test. 
7. Verify only the selected test runs. 

Verify the following package scripts continue to function as expected: 
  - `test:native`
  - `test:native:clean`
  - `test:native:debug`
  - `test:native:perf`
  - `test:native:perf:baseline`
  - `test:native:update`

### Testing Instructions for Keyboard

n/a

## Screenshots or screencast 
n/a
